test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,20 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+try {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, function() {
+    throw new Error('fs.open callback should not have been called');
+  });
+
+  throw new Error('fs.open did not throw despite bad flags');
+} catch (err) {
+  if (!(err instanceof TypeError)) {
+    throw new Error('unexpected error: ' + err);
+  }
+}
```

Comment:
I second brendan's suggestion. What if TypeError is thrown because of some other problem? I prefer checking the actual message.

I would say, when the mesaages are updated, let the tests also be updated.